src/components/TextField/TextField.tsx

```diff
@@ -37,11 +40,26 @@ class TextField extends React.Component<TextFieldProps & DebouncedOnChangePrivat
         suffix={this.props.suffix}
         underlined={this.props.underlined}
         onChanged={this.props.unifiedOnChange}
+        componentRef={this.getRef}
         {...getBaseTextFieldProps(this.props)}
       />
     );
   }
 
+  public focus = () => {
+    const el = this.fabricTextFieldElement;
+    if (el) {
+      const valueLength = this.props.value ? this.props.value.length : 0;
+
+      el.focus();
+      el.setSelectionRange(valueLength, valueLength);
+    }
+  };
+
+  private getRef = (ref: ITextField | null) => {
+    this.fabricTextFieldElement = ref ? ref : undefined;
```

Comment:
this could be slightly simplified to

`this.fabricTextFieldElement = ref || undefined;`

Reply:
See the other comment below. The subtle difference is the return types.  Your suggestion returns `ITextField | null` and the existing one is `ITextField | undefined`.